### PR TITLE
Update to Cadence v0.39.1

### PR DESCRIPTION
**Pull request by turbolent**
Update to the latest Cadence version v0.39.1.

Includes the update to the latest major version, [v0.39.0](https://github.com/onflow/cadence/releases/tag/v0.39.0). 
That release introduces the new Capability Controller API. Enable it on all networks, except for Mainnet.

Thanks to @bluesign for fixing up al the integration tests using the Emulator!

### CODE REVIEW

**Comment by turbolent** [comment]
@janezpodhostnik @pattyshack Could you please take care of implementing the new account ID generation function?

File: fvm/fvm_test.go [hunk 1]
```diff
@@ -2448,3 +2448,71 @@ func TestAttachments(t *testing.T) {
...
+			) {
+				txBody := flow.NewTransactionBody().
+					SetScript([]byte(`
+						transaction {
+                          prepare(signer: AuthAccount) {
+							let cap = signer.capabilities.storage.issue<&Int>(/storage/foo)
+                            assert(cap.id == 1)
+
+                            let cap2 = signer.capabilities.storage.issue<&String>(/storage/bar)
+                            assert(cap2.id == 2)
```
Comment (janezpodhostnik) [inline fvm/fvm_test.go:2481]: thank you for adding a test for this!

File: fvm/fvm_test.go [hunk 2]
```diff
@@ -2448,3 +2448,71 @@ func TestAttachments(t *testing.T) {
...
+				t *testing.T,
+				vm fvm.VM,
+				chain flow.Chain,
+				ctx fvm.Context,
+				snapshotTree snapshot.SnapshotTree,
+			) {
+				txBody := flow.NewTransactionBody().
+					SetScript([]byte(`
+						transaction {
+                          prepare(signer: AuthAccount) {
```
Comment (janezpodhostnik) [inline fvm/fvm_test.go:2476]: The formatting on the cadence code is a bit weird, probably a mix of tabs and spaces.

**Review by janezpodhostnik — APPROVED** [review]

File: fvm/fvm_test.go (hunk 2 above)
Comment (turbolent) [reply fvm/fvm_test.go]: Hopefull fixed in 6af91b03b2bbd078cf8a288b24fa86149890e959

**Review by zhangchiqing — APPROVED** [review]

**Review by SupunS — APPROVED** [review]

**Comment by turbolent** [comment]
All the integration tests using the Emulator are broken, because the APIs change somehow :-/
@janezpodhostnik @bluesign could you please advise on how to adjust the code?

**Comment by bluesign** [comment]
> All the integration tests using the Emulator are broken, because the APIs change somehow :-/
> @janezpodhostnik @bluesign could you please advise on how to adjust the code?

I can fix them, just no idea how to run them locally.

**Comment by turbolent** [comment]
@bluesign That would be much appreciated! You can run the tests with `go test --tags relic ./...` (note the build tag `relic` must be enabled)

**Comment by turbolent** [comment]
@jordanschalm Could you or someone else who has knowledge about the integration tests using the Emulator please review the changes to them?

**Review by jordanschalm — APPROVED** [review]

**Comment by turbolent** [comment]
bors merge

**Comment by turbolent** [comment]
bors retry

**Comment by turbolent** [comment]
bors retry

**Pull request merged**